discovery-infra/test_infra/controllers/node_controllers/node.py

```diff
@@ -145,3 +145,21 @@ def attach_test_disk(self, disk_size, bootable=False):
 
     def detach_all_test_disks(self):
         self.node_controller.detach_all_test_disks(self.name)
+
+    def attach_interface(self, network_xml, target_interface="vnet3"):
```

Comment:
I am not sure the default target interface should be defined on the node controller level.

Reply:
its not in the node controller, its in node :)